### PR TITLE
#8694: add --collect-only for sweep tests

### PR DESCRIPTION
**Pull request by sjameelTT**

### CODE REVIEW

**Comment by sjameelTT** [comment]
Sweep failure is due to the floating point exception on matmul.

File: tests/ttnn/sweep_tests/sweeps/__init__.py [hunk 1]
```diff
@@ -199,6 +199,14 @@ def name_to_string(name):
     return table_name
 
 
+def collect_only(*, include):
+    for file_name in sorted(SWEEP_SOURCES_DIR.glob("**/*.py")):
+        sweep_name = get_sweep_name(file_name)
+        if include and sweep_name not in include:
+            continue
+        print(sweep_name)
```
Comment (TT-BrianLiu) [inline tests/ttnn/sweep_tests/sweeps/__init__.py:207]: Can we refactor this into a generator function and have `run_sweeps` just call this?

Comment (sjameelTT) [reply tests/ttnn/sweep_tests/sweeps/__init__.py:207]: Sure

**Review by xanderchin — APPROVED** [review]

**Comment by sjameelTT** [comment]
Sweep test failure is the same as the one on main. Any issues with this change would prevent sweep tests from running at all, so it should not be a problem.

**Pull request merged**